src/CarbonAware.WebApi/test/integrationTests/IntegrationTestingBase.cs

```diff
@@ -0,0 +1,120 @@
+﻿using CarbonAware.DataSources.Configuration;
+using Microsoft.AspNetCore.Mvc.Testing;
+using NUnit.Framework;
+using System.Net.Http.Headers;
+using System.Net.Mime;
+using System.Text.Json;
+using System.Web;
+
+namespace CarbonAware.WebApi.IntegrationTests;
+
+/// <summary>
+/// A base class that does all the common setup for the Integration Testing
+/// Overrides WebAPI factory by switching out different configurations via _datasource
+/// </summary>
+public abstract class IntegrationTestingBase
+{
+    internal DataSourceType _dataSource;
+    internal string? _dataSourceEnv;
+    internal WebApplicationFactory<Program> _factory;
+    protected HttpClient _client;
+    protected IDataSourceMocker _dataSourceMocker;
+
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    public IntegrationTestingBase(DataSourceType dataSource)
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    {
+        _dataSource = dataSource;
+        _dataSourceEnv = null;
+        _factory = new WebApplicationFactory<Program>();
+    }
+
+
+    public string ConstructDateQueryURI(string Url, string location, DateTimeOffset start, DateTimeOffset end)
+    {
+        // Use HTTP Query builder
+        var builder = new UriBuilder();
+
+        //Add all query parameters
+        var query = HttpUtility.ParseQueryString(builder.Query);
+        query["locations"] = location;
+        query["time"] = $"{start:O}";
+        query["toTime"] = $"{end:O}";
+
+        //Generate final query string
+        builder.Query = query.ToString();
+        builder.Path = Url;
+
+        //These values are blank as they are set by the SDK
+        builder.Scheme = "";
+        builder.Port = -1;
+        builder.Host = "";
+
+        return builder.ToString();
+    }
+
+    public async Task<HttpResponseMessage?> PostJSONBodyToURI(object body, string URI)
+    {
+        var jsonBody = JsonSerializer.Serialize(body);
+        StringContent _content = new StringContent(jsonBody);
+
+        var mediaType = new MediaTypeHeaderValue(MediaTypeNames.Application.Json);
```

Comment:
nit: This `mediaType` can be part an attribute of the class, no need to create a new one always.

Reply:
yes, re: the webapi.xml -- we just removed it, and added to .gitignore